File: BookStore.BuisneLogic/Extensions/UesrsFilterExtensions/UsersFilterModel.cs
using BookStore.BusinessLogic.Models.UesrsFilterModel;

namespace BookStore.BusinessLogic.Extensions.UesrsFilterExtensions
{
    public static class UsersFilterModelBLMapToUsersFilterModelDA
    {
        public static DataAccess.Models.UesrsFilterModel.UsersFilterModel Map(this UsersFilterModel usersFilterModel)
        {
            return new BookStore.DataAccess.Models.UesrsFilterModel.UsersFilterModel
            {
                PageCount = usersFilterModel.PageCount,
                PageSize = usersFilterModel.PageSize,
                UserStatus = (DataAccess.Models.Enums.Enums.UserStatus)usersFilterModel.UserStatus,
                SearchString = usersFilterModel.SearchString,
                SortType = (DataAccess.Models.Enums.Enums.UserSortType)usersFilterModel.SortType,
                SortingDirection = (DataAccess.Models.Enums.Enums.SortingDirection)usersFilterModel.SortingDirection                
            };
        }
    }
}

File: StoreClient/src/app/shared/models/Users/users-filter-model.ts
import {UserSortType} from 'src/app/shared/enums/user-sort-type';
import {UserStatus} from 'src/app/shared/enums/user-status';
import {SortingDirection} from 'src/app/shared/enums/sorting-direction';

export class UsersFilterModel {
  sortType: UserSortType;
  userStatus: UserStatus;
  sortingDirection: SortingDirection;
  searchString: string;
  pageCount: number;
  pageSize: number;
}
